torchtune/modules/transformer.py

```diff
@@ -460,8 +465,14 @@ def forward(
         # shape: [b, s, d]
         h = self.norm(h)
 
-        # shape: [b, s, out_dim] - out_dim is usually the vocab size
-        output = self.output(h).float()
+        # shape: [b, seq_len//num_chunks, out_dim] - out_dim is usually the vocab size
+        if self.num_output_chunks > 0:
+            # chunk for ChunkedCrossEntropyLoss, saving memory
+            output = [
+                self.output(chunk) for chunk in h.chunk(self.num_output_chunks, dim=1)
+            ]
+        else:
+            output = self.output(h)
```

Comment:
Sorry if we already discussed this, but why move the float call from here to the recipe? Easy for recipe writers to trip up and doesnt seem like its conceptually different?

Reply:
thats true! I can put the float back here. It doesnt need to be in the recipe.